### ISSUE TITLE
setting up Prometheus for Lambda deployments

### ISSUE THREAD

**Issue by jjmachan** [bentoml/aws-lambda-deploy#24]
Currently, with the `BentoML==1.0.0a4` release, the lambda deployment is broken due because Prometheus `PROMETHEUS_MULTIPROC_DIR` is not being setup correctly.

Error log from CloudWatch
```
[ERROR] AssertionError
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/bentoml/bento/app.py", line 15, in <module>
    setattr(this_module, api_name, Mangum(bento_service.asgi_app))
  File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/service/service.py", line 156, in asgi_app
    return ServiceAppFactory(self)()
  File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/service_app.py", line 316, in __call__
    app = super().__call__()
  File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/base_app.py", line 61, in __call__
    return Starlette(
  File "/opt/conda/lib/python3.9/site-packages/starlette/applications.py", line 71, in __init__
    self.middleware_stack = self.build_middleware_stack()
  File "/opt/conda/lib/python3.9/site-packages/starlette/applications.py", line 98, in build_middleware_stack
    app = cls(app=app, **options)
  File "/opt/conda/lib/python3.9/site-packages/simple_di/__init__.py", line 124, in _
    return func(*_inject_args(bind.args), **_inject_kwargs(bind.kwargs))
  File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/instruments.py", line 37, in __init__
    self.metrics_request_duration = metrics_client.Histogram(
  File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/metrics/prometheus.py", line 106, in Histogram
    self.prometheus_client.Histogram,
  File "/opt/conda/lib/python3.9/site-packages/bentoml/_internal/server/metrics/prometheus.py", line 48, in prometheus_client
    assert os.path.isdir(self.multiproc_dir)
 ```

but more importantly, using lambda services with Prometheus and logging the metrics properly is something that needs some more thought into
Interesting Resources
1. https://aws.amazon.com/cn/blogs/opensource/aws-lambda-metrics-support-for-amazon-managed-service-for-prometheus-now-available-in-aws-distro-for-opentelemetry/

**Comment by camille-vanhoffelen** [comment]
I still get this exact error when deploying to aws-lambda despite my `~/bentoctl/operators/aws-lambda` containing pull #25 mentioned above.

bentoctl, version 1.0.0a4
bentoml, version 1.0.0a4

**Comment by jjmachan** [comment]
can you try with bentoml 1.0.0a5, it should be working now

**Comment by camille-vanhoffelen** [comment]
Tried with bentoml 1.0.0a5, and ran into a different error - but seems related to the use of the `latest` tag. FYI this setup works fine with `aws-ec2` deployment. Here's the stacktrace:

[stacktrace.txt](https://github.com/bentoml/aws-lambda-deploy/files/8215185/stacktrace.txt)
.

**Comment by camille-vanhoffelen** [comment]
FYI both problems solved with bentoml 1.0.0a6

**Comment by jjmachan** [comment]
We now have a permanent fix with #44. The /metrics is disabled via the attached bentoml runtime config file. For tracking the metrics you can use the aws-cloudwatch service that is configured with the deployment.

Feel free to open a new issue if you face any other problems
